### PR TITLE
feat: [OSM-347] Next `csharp` bundle attempt

### PR DESCRIPTION
**Pull request by dotkas**
Ok, my previous [attempt](https://github.com/snyk/snyk-nuget-plugin/pull/175) was not really thought through.

Using this package as a CLI Plugin makes the plugin not know where it exists during runtime, causing it to not to be able to find the .cs files in its own dist folder. Should have thought about it.

Second best solution is keeping C# code in memory and writing it to a temporary folder. We already do this heavily in the tests, so it is a trivial thing to change.

Not super satisfied with the outcome, but I did not see a better way. There are some Github projects to execute .NET code from Typescript, but it's a hassle.

In short:
* Refactored the fixture test helpers to be used by actual non-test code
* Removed the raw C# code
* Re-used the test fixture helper to write the C# code to a temporary file store
* Added a simple memory cache to avoid tons of `dotnet restore` and spamming of `/tmp` when running tests and in the future when scanning multiple `<TargetFramework>s`.

### CODE REVIEW

File: lib/nuget-parser/csharp/generator.ts
Comment (dotkas) [inline lib/nuget-parser/csharp/generator.ts:1]: Basically just moved this from the test fixture helpers to here, didn't change any logic, really.

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "test": "npm run test:unit",
     "test:unit": "jest --coverage --runInBand",
     "lint": "npm run lint:prettier && npm run lint:eslint",
-    "lint:eslint": "eslint . --ext .ts",
+    "lint:eslint": "eslint -c .eslintrc '{lib,test}/**/*.ts'",
```
Comment (dotkas) [inline package.json:10]: CI pipeline was failing due to linting errors for node_modules in `dist` 🤷🏻

File: tsconfig.json [hunk 2]
```diff
@@ -5,12 +5,13 @@
     "target": "es2019",
     "module": "commonjs",
     "sourceMap": true,
+    "declaration": true,
```
Comment (dotkas) [inline tsconfig.json:8]: Not really relevant, but `package.json` referred to it.

**Review by orsagie — APPROVED** [review]
